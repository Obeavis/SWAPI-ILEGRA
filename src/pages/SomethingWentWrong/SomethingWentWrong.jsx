import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import somethingwentwrong from "../../assets/images/something-went-wrong.png";
import "./SomethingWentWrong.scss"

const SomethingWentWrong = () => (
	<Container className="something-went-wrong">
		<Row>
			<Col lg={6} className="text-center text-lg-right">
				<h2 className="text-20 text-bold">Ooops</h2>

				<p className="text-white text-12">
					something went wrong
				</p>
			</Col>
			<Col lg={6} className="text-center text-lg-left">
				<img
					alt="stormtrooper"
					src={somethingwentwrong}
					className="img-fluid"
				/>
			</Col>
		</Row>
	</Container>
)

export default SomethingWentWrong;