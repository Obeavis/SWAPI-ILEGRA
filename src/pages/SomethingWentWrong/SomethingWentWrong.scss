.something-went-wrong {
  margin-top: 150px;

  h2 {
    color: #f2bc00;
  }

  img {
    margin-left: auto;
    margin-right: auto;
    margin-top: 1rem;
  }
}
