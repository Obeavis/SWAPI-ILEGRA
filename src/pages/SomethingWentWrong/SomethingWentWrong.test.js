import React from "react";
import { MemoryRouter } from 'react-router-dom';
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import SomethingWentWrong from "./SomethingWentWrong";

configure({ adapter: new Adapter() });

describe("Film Component", () => {

    const component = (
        <MemoryRouter>
                <SomethingWentWrong />
        </MemoryRouter>
    )

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it("Renders the right image", () => {
        const wrapper = mount(component);
        const image = wrapper.find('img').prop("src");

        expect(image).toEqual('something-went-wrong.png');

    });

    it("Have the right text inside", () => {
        const wrapper = mount(component);
        const error = wrapper.find('h2').text();
        const text = wrapper.find('p').text();

        expect(error).toEqual('Ooops');
        expect(text).toEqual("something went wrong");
    });

})