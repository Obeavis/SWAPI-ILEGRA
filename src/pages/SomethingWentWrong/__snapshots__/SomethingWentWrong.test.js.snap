// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Component matches the snapshot 1`] = `
<div
  className="something-went-wrong container"
>
  <div
    className="row"
  >
    <div
      className="text-center text-lg-right col-lg-6"
    >
      <h2
        className="text-20 text-bold"
      >
        Ooops
      </h2>
      <p
        className="text-white text-12"
      >
        something went wrong
      </p>
    </div>
    <div
      className="text-center text-lg-left col-lg-6"
    >
      <img
        alt="stormtrooper"
        className="img-fluid"
        src="something-went-wrong.png"
      />
    </div>
  </div>
</div>
`;
